ambry-router/src/main/java/com/github/ambry/router/DeleteOperation.java

```diff
@@ -284,10 +285,11 @@ private void cleanupExpiredInflightRequests(
         onErrorResponse(requestInfo.getReplicaId(),
             RouterUtils.buildTimeoutException(correlationId, requestInfo.getReplicaId().getDataNodeId(), blobId));
         requestRegistrationCallback.registerRequestToDrop(correlationId);
-      } else {
-        // the entries are ordered by correlation id and time. Break on the first request that has not timed out.
-        break;
       }
+      // Note: Even though the entries are ordered by correlation id and their creation time, we cannot break out of
```

Comment:
Can this have any performance effect?

Reply:
Same question. This change makes us loop all requests

Reply:
I was just wondering if it might be okay for now size of the Map is currently fixed/small since we can have probably maximum of 3 requestInfos (determined by the router request parallelism factor) for each operation at a time  or may be more if we use adaptive operation tracker.